processor/transformprocessor/internal/traces/processor_test.go

```diff
@@ -0,0 +1,238 @@
+			query: `set(status.code, 1) where attributes["http.path"] == "/health"`,
+			want: func(td pdata.Traces) {
+				td.ResourceSpans().At(0).InstrumentationLibrarySpans().At(0).Spans().At(0).Status().SetCode(pdata.StatusCodeOk)
+				td.ResourceSpans().At(0).InstrumentationLibrarySpans().At(0).Spans().At(1).Status().SetCode(pdata.StatusCodeOk)
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.query, func(t *testing.T) {
+			td := constructTraces()
+			statements, err := parse([]string{tt.query})
+			assert.NoError(t, err)
+			processor, err := NewProcessor(statements, component.ProcessorCreateSettings{})
+			assert.NoError(t, err)
+
+			_, err = processor.ProcessTraces(context.Background(), td)
+			assert.NoError(t, err)
+
+			exTd := constructTraces()
+			tt.want(exTd)
+
+			assert.Equal(t, exTd, td)
+		})
+	}
+}
+
+func BenchmarkTwoSpans(b *testing.B) {
+	tests := []struct {
+		name    string
+		queries []string
+	}{
+		{
+			name:    "no processing",
+			queries: []string{},
+		},
+		{
+			name:    "set attribute",
+			queries: []string{`set(attributes["test"], "pass") where name == "operationA"`},
+		},
+		{
+			name:    "keep attribute",
+			queries: []string{`keep(attributes, "http.method") where name == "operationA"`},
+		},
+		{
+			name:    "no match",
+			queries: []string{`keep(attributes, "http.method") where name == "unknownOperation"`},
+		},
+		{
+			name:    "inner field",
+			queries: []string{`set(status.code, 1) where attributes["http.path"] == "/health"`},
+		},
+		{
+			name: "inner field both spans",
+			queries: []string{
+				`set(status.code, 1) where name == "operationA"`,
+				`set(status.code, 2) where name == "operationB"`,
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		b.Run(tt.name, func(b *testing.B) {
+			statements, err := parse(tt.queries)
+			assert.NoError(b, err)
+			b.ResetTimer()
+			for n := 0; n < b.N; n++ {
+				td := constructTraces()
+				process(td, statements)
+			}
+		})
+	}
+}
+
+func BenchmarkHundredSpans(b *testing.B) {
+	tests := []struct {
+		name    string
+		queries []string
+	}{
+		{
+			name:    "no processing",
+			queries: []string{},
+		},
+		{
+			name: "set status code",
+			queries: []string{
+				`set(status.code, 1) where name == "operationA"`,
+				`set(status.code, 2) where name == "operationB"`,
+			},
+		},
+		{
+			name: "hundred statements",
+			queries: func() []string {
+				queries := make([]string, 0)
+				queries = append(queries, `set(status.code, 1) where name == "operationA"`)
+				for i := 0; i < 99; i++ {
+					queries = append(queries, `keep(attributes, "http.method") where name == "unknownOperation"`)
+				}
+				return queries
+			}(),
+		},
+	}
+	for _, tt := range tests {
+		b.Run(tt.name, func(b *testing.B) {
+			statements, err := parse(tt.queries)
+			assert.NoError(b, err)
+			b.ResetTimer()
+			for n := 0; n < b.N; n++ {
+				td := constructTracesNum(100)
+				process(td, statements)
+			}
+		})
+	}
+}
+
+func constructTraces() pdata.Traces {
```

Comment:
We have a "helper" package for this https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/internal/coreinternal/testdata